sql/wsrep_sst.cc

```diff
@@ -498,8 +529,21 @@ static ssize_t sst_prepare_other (const char*  method,
     return (ret < 0 ? ret : -EMSGSIZE);
   }
 
+  wsp::env env(NULL);
+  if (env.error())
+  {
+    WSREP_ERROR("sst_prepare_other(): env. var ctor failed: %d", -env.error());
+    return -env.error();
+  }
+
+  if ((ret= sst_append_auth_env(env, sst_auth)))
+  {
+    WSREP_ERROR("sst_prepare_other(): appending auth failed: %d", ret);
+    return ret;
+  }
+
   pthread_t tmp;
-  sst_thread_arg arg(cmd_str);
+  sst_thread_arg arg(cmd_str(), env());
```

Comment:
Not quite part of this patch, but below return value from ```pthread_create()``` is returned directly. According to http://man7.org/linux/man-pages/man3/pthread_create.3.html the error returned is error number (positive integer). However, all other error handling in this function return negative error values.